File: Algorithms/504. Base 7/javascript/solution.iteration.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function (num) {
    if (num === 0) return '0';
    let res = '';
    let sign = num < 0 ? '-' : '';
    num = Math.abs(num);
    while (num > 0) {
        res = (num % 7) + res;
        num = Math.floor(num / 7);
    }
    return sign + res;
};

File: Algorithms/504. Base 7/python/solution.iteration.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return '0'

        n, ans = abs(num), ''
        while n:
            ans = str(n % 7) + ans
            n //= 7
        return '-' * (num < 0) + ans
